trunk/NDHMS/MPP/mpp_land.F

```diff
@@ -898,6 +899,38 @@ subroutine write_IO_RT_int (in_buff,out_buff)
       return
   end subroutine write_IO_RT_int
 
+  subroutine write_IO_RT_int8(in_buff,out_buff)
+      ! the IO node will receive the data from the rest process.
+      integer(kind=int64),intent(in),dimension(:,:) :: in_buff
+      integer(kind=int64),intent(out),dimension(:,:) ::  out_buff
+      integer tag, i, status, ierr,size
+      integer ibegin,iend,jbegin,jend
+      if(my_id .ne. IO_id) then
+          size = local_rt_nx*local_rt_ny
+          tag = 2
+          call mpi_send(in_buff,size,MPI_INTEGER, IO_id,     &
```

Comment:
I figured out why the compiler isn't throwing any warning with the type-mismatch. The `mpi` module is used and it is the `mpi_f08` module that uses the MPI_Datatypes. In the `mpi` module the type of the datatype is just an integer and the compiler can't even tell at compile time the difference between `MPI_INTEGER` and `MPI_COMPLEX`. Transitioning to `use mpi_f08` would be worthwhile but an arduous process and will have to be left to some future date.

Reply:
It might be worth trying it and seeing just how much compiler spew we get from it. If it's not too bad, adding type-checking would definitely be worthwhile...

Reply:
There might be some repeat counting (I just used `grep` and `wc -l`) but it looks like around 140 errors to fix.

I can create an issue talking about the transition from `mpi` to `mpi_f08`. I think it is worth it but would require editing almost every single MPI subroutine call. This would require some planning and a chunk of time for edits. A good example of what would need to be changed comes from looking at the documentation for [MPI_Send](https://www.open-mpi.org/doc/v4.0/man3/MPI_Send.3.php), which would require changing the communicator and type arguments from integers to MPI datatypes. It's nice the communication is handled in `MPP_Land`, which helps lower the amounts of changes needed, but there were type errors in other files as well.

Good news is that I looking over the code I found only two more places in `mpp_land.F` that needed to be changed to `MPI_INTEGER8`.